openml-lightgbm/lightgbm-builder/make.sh

```diff
@@ -18,13 +18,14 @@
 
 set -e
 
-VERSION="$1"
+LIGHTGBM_VERSION="$1"
+LIGHTGBMLIB_VERSION="$2"
```

Comment:
So, from [make-lightgbm](https://github.com/feedzai/make-lightgbm/blob/master/make.sh). `LIGHTGBMLIB_VERSION` is the package version. 

But I don't know why is the value of `LIGHTGBM_VERSION` `fc79b366153e8aa5cf2996b4b753e28e5a75e878`? Is it generated?

Reply:
That's the git commit id of LightGBM. make-lightgbm will fetch that instead of the latest release which has the tag v2.3.1.

Reply:
If one day they rebased or update all the commits, the hash of the commit might change as well.
If that happens we would need to find the hash again. So, Don't we have other way to get it?

BTW, in that case, we shouldn't name it as `LIGHTGBMLIB_VERSION`, instead `LIGHTGBMLIB_COMMIT_HASH`

Reply:
We already do so. But it is the "version" for a lack of a better name and not the hash. Imagine they changed the tag v2.3.1 to point to another commit: the commit changes, but it's still what they call v2.3.1.

If that happens, make-lightgbm checks the hash of that tag and if it doesn't match __commit_id__ in build/ triggers a new build. If it is the same nothing happens.

Reply:
Guys, it's really cool the level of critical thinking here, but I guess that we need to establish some ground assumptions here. Feel free to question my statements below:

- Rebases should only be done over development branches. The master branch (as well as hotfix branches) should be monotonic, and are usually marked as protected (i.e. cannot be replayed/rebased). **Referring to commits in protected branches should be safe.**
- Versions should also be immutable: if they are attached to a given commit, then must remain so, otherwise you'll end up with two users, both using version v2.3.1, which are in fact using different versions of the product (i.e. it invalidates the concept of version). **Referring to versions (i.e. git tags) should be safe.**
- I agree with Sheng's suggestion to change the variable name, to make it more consistent 👍

Reply:
I have no problems choosing another name.

Just to be sure, the request is to rename `LIGHTGBM_VERSION` to `LIGHTGBMLIB_COMMIT_HASH` even if it receives a tag name / version like v2.3.1?

What about `LIGHTGBM_CHECKOUT_TARGET`?